test/unit/alphabet/composite/alphabet_variant_detail_test.cpp

```diff
@@ -123,3 +124,48 @@ TEST(alphabet_variant_detail_test, char_to_rank)
     EXPECT_EQ(char_to_rank4['G'], 2);
     EXPECT_EQ(char_to_rank4['T'], 4);
 }
+
+TEST(alphabet_variant_detail_test, first_valid_char_table)
+{
+    constexpr std::array first_valid_char_table2 = detail_alphabet_variant<seqan3::dna4,
+                                                                           seqan3::gap>::first_valid_char_table;
+    EXPECT_EQ(first_valid_char_table2.size(), 256u);
+    EXPECT_EQ(first_valid_char_table2['A'], 0);
+    EXPECT_EQ(first_valid_char_table2['C'], 0);
+    EXPECT_EQ(first_valid_char_table2['G'], 0);
+    EXPECT_EQ(first_valid_char_table2['T'], 0);
+    EXPECT_EQ(first_valid_char_table2['-'], 1);
+    EXPECT_EQ(first_valid_char_table2['X'], 2); // invalid char
```

Comment:
Do the number mean, to which alphabet in the table the character belongs to? And if the number is higher than the table dimension, its not a valid character?

Reply:
Yeah, kind of. Except when two or more alphabets have the same "valid" char, it chooses the first one.